presto-orc/src/test/java/com/facebook/presto/orc/LongDictionaryProviderTest.java

```diff
@@ -0,0 +1,291 @@
+                ++index;
+            }
+        }
+        return result;
+    }
+
+    @DataProvider(name = "dataForTestDataCorruptionCheck")
+    public Object[][] dataForTestDataCorruptionCheck()
+    {
+        return flattenDataCorruptionCheckTestInput(dataForDictionaryLoadingTests());
+    }
+
+    @Test(expectedExceptions = OrcCorruptionException.class, expectedExceptionsMessageRegExp = ".* Retrieving non-long dictionary types from LongDictionaryProvider.*")
+    public void testTypeMismatchedRequest()
+            throws IOException
+    {
+        StreamId streamId = new StreamId(2, 0, DICTIONARY_DATA);
+
+        DynamicSliceOutput sliceOutput = createStringSliceOutput(streamId);
+        ValueInputStream<?> valueStream = createValueStream(sliceOutput.slice());
+        StreamCheckpoint streamCheckpoint = getDictionaryStreamCheckpoint(streamId, OrcType.OrcTypeKind.BINARY, ColumnEncoding.ColumnEncodingKind.DICTIONARY);
+        InputStreamSource<?> streamSource = createCheckpointStreamSource(valueStream, streamCheckpoint);
+
+        LongDictionaryProvider longDictionaryProvider = new LongDictionaryProvider(new InputStreamSources(ImmutableMap.of(streamId, streamSource)));
+        long[] dictionary = new long[0];
+        longDictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), dictionary, 10);
+    }
+
+    private static class NodeId
+    {
+        private final int node;
+        private final int sequence;
+
+        public NodeId(int node, int sequence)
+        {
+            this.node = node;
+            this.sequence = sequence;
+        }
+
+        public StreamId toDictionaryDataStreamId()
+        {
+            return new StreamId(node, sequence, DICTIONARY_DATA);
+        }
+    }
+
+    private static class DummyOrcDataSource
+            implements OrcDataSource
+    {
+        @Override
+        public OrcDataSourceId getId()
+        {
+            return ORC_DATA_SOURCE_ID;
+        }
+
+        @Override
+        public long getReadBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getReadTimeNanos()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getSize()
+        {
+            return 0;
+        }
+
+        @Override
+        public void readFully(long position, byte[] buffer)
+        {
+        }
+
+        @Override
+        public void readFully(long position, byte[] buffer, int bufferOffset, int bufferLength)
+        {
+        }
+
+        @Override
+        public <K> Map<K, OrcDataSourceInput> readFully(Map<K, DiskRange> diskRanges)
+        {
+            return null;
+        }
+    }
+
+    StreamDescriptor createFlatStreamDescriptor(StreamId streamId)
+    {
+        return new StreamDescriptor("test_dictionary_stream", streamId.getColumn(), String.format("field_%d", streamId.getColumn()), LONG_TYPE, DUMMY_ORC_DATA_SOURCE, ImmutableList.of(), streamId.getSequence());
```

Comment:
It doesn't have to be unique? If, so just slap some fake constant.
Otherwise doing `field_" + streamId.getColumn()` concat or Integer.toString(streamId.getColumn()) would be just fine